backend/fms_core/serializers.py

```diff
@@ -591,12 +593,22 @@ def get_blocked_status_count(self, obj):
     def get_latest_release_update(self, obj):
         return DatasetFile.objects.filter(readset__dataset=obj.id).aggregate(Max("release_status_timestamp"))["release_status_timestamp__max"]
 
+class ReadsetSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Readset
+        fields = ("id", "name", "dataset", "sample_name", "derived_sample")
+
 class DatasetFileSerializer(serializers.ModelSerializer):
-    dataset = serializers.IntegerField(read_only=True, source='readset.dataset.id')
-    sample_name = serializers.CharField(read_only=True, source='readset.sample_name')
+    readset = serializers.SerializerMethodField()
```

Comment:
you can just plug in readset = ReadsetSerializer(read_only=True) here i think instead of using a method.